src/rdkafka_broker.c

```diff
@@ -344,6 +344,7 @@ void rd_kafka_broker_set_state(rd_kafka_broker_t *rkb, int state) {
                             rd_atomic32_get(
                                 &rkb->rkb_rk->rk_logical_broker_cnt) &&
                     !rd_kafka_terminating(rkb->rkb_rk)) {
+                        rd_kafka_rebootstrap(rkb->rkb_rk);
```

Comment:
Ideally we should be removing all the other brokers. We are rebootstraping and we don't need old brokers. We will be sending metadata request to these new bootstrap brokers.

This can be an improvement for now maybe.

Reply:
The metadata response should be the controlling the brokers that are removed alone, in case we rebootstrap when we'll reconnect and find the new metadata the old ones will be removed but until then we assume nothing has changed.